src/sql/connection/connection.py

```diff
@@ -484,95 +257,163 @@ def close_connection_with_descriptor(cls, descriptor):
             cls.connections.pop(
                 str(conn.metadata.bind.url) if IS_SQLALCHEMY_ONE else str(conn.url)
             )
+            conn.connection.close()
 
-        conn.close()
+    @classmethod
+    def connections_table(cls):
+        """Returns the current connections as a table"""
+        connections = cls._get_connections()
 
-    def close(self):
-        """Close the current connection"""
-        for rs in self._result_sets:
-            rs._sqlaproxy.close()
+        def map_values(d):
+            d["current"] = "*" if d["current"] else ""
+            d["alias"] = d["alias"] if d["alias"] else ""
+            return d
 
-        self.session.close()
+        return display.ConnectionsTable(
+            headers=["current", "url", "alias"],
+            rows_maps=[map_values(c) for c in connections],
+        )
 
     @classmethod
-    def close_all(cls, verbose=False):
-        """Close all connections"""
-        connections = Connection.connections.copy()
-        for name, conn in connections.items():
-            conn.close()
+    def from_connect_str(
+        cls, connect_str=None, connect_args=None, creator=None, alias=None
+    ):
+        """Creates a new connection from a connection string"""
+        connect_args = connect_args or {}
 
-            if verbose:
-                display.message(f"Closing {name}")
+        try:
+            if creator:
+                engine = sqlalchemy.create_engine(
+                    connect_str,
+                    connect_args=connect_args,
+                    creator=creator,
+                )
+            else:
+                engine = sqlalchemy.create_engine(
+                    connect_str,
+                    connect_args=connect_args,
+                )
+        except (ModuleNotFoundError, NoSuchModuleError) as e:
+            suggestion_str = get_missing_package_suggestion_str(e)
+            raise exceptions.MissingPackageError(
+                "\n\n".join(
+                    [
+                        str(e),
+                        suggestion_str,
+                        PLOOMBER_DOCS_LINK_STR,
+                    ]
+                )
+            ) from e
+        except Exception as e:
+            raise cls._error_invalid_connection_info(e, connect_str) from e
 
-        cls.connections = {}
+        connection = SQLAlchemyConnection(engine, alias=alias)
```

Comment:
What happens if we pass a non-SQL alchemy connection string here? i.e. questdb.

Reply:
it'll fail. if you use `from_connect_str` and pass an invalid sqlalchemy string, it'll throw an error because connection strings assume you're using sqlalchemy connections